packages/x-components/src/x-modules/experience-controls/store/module.ts

```diff
@@ -3,4 +3,25 @@
  *
  * @internal
  */
-export const experienceControlsXStoreModule: any = {};
+export const experienceControlsXStoreModule: any = {
+  state: () => ({
+    config: {
+      numberOfCarousels: 5,
+      resultsPerCarousels: 12
```

Comment:
This is coupled to the information that we know the endpoint has. What will happen when the endpoint adds  a new field? We would have to update the type of the state to represent the new field.

The state doesn't know the specific configuration options that it will have to store, it only knows it needs to have a place where to put it (the field I talked about in a comment below).

I'm imagining the state of this module to have a type similar to:

````javascript
{
    controls: { [key: string]: unknown },
    events: Partial<XEventsTypes>
...
}

````